gobblin-core/src/main/java/gobblin/publisher/HiveRegistrationPublisher.java

```diff
@@ -125,10 +155,11 @@ public void publishData(Collection<? extends WorkUnitState> states) throws IOExc
 
         final HiveRegistrationPolicy policy = HiveRegistrationPolicyBase.getPolicy(taskSpecificState);
         for ( final String path : state.getPropAsList(ConfigurationKeys.PUBLISHER_DIRS) ) {
-          if (pathsToRegisterFromSingleState.contains(path)){
+          if (pathsToRegisterFromSingleState.contains(path) && isDefaultPathDedupeEnaled ){
```

Comment:
Check boolean config first before checking the set to be more optimal when dedupe is disabled.